webrender/src/render_backend.rs

```diff
@@ -620,9 +620,22 @@ impl RenderBackend {
             FrameMsg::Scroll(delta, cursor) => {
                 profile_scope!("Scroll");
 
-                let should_render = doc.scroll(delta, cursor)
-                    && doc.render_on_scroll == Some(true);
+                let mut should_render = true;
+                let node_index = match doc.hit_tester {
+                    Some(ref hit_tester) => {
+                        let test = HitTest::new(None, cursor, HitTestFlags::empty());
```

Comment:
since this is where we have the node index and this is where checking for `scroll_nearest_scrolling_ancestor` is really needed, can we rewrite this part as:
```rust
let should_render = match doc.hit_tester {
...
};
```

Reply:
This bit of ugliness is necessary to avoid a double-borrow of doc, unfortunately. I can leave a comment about that.